lib/fluent/plugin/kubernetes_metadata_watch_pods.rb

```diff
@@ -25,8 +25,20 @@ module WatchPods
 
     def start_pod_watch
       begin
-        resource_version = @client.get_pods.resourceVersion
-        watcher          = @client.watch_pods(resource_version)
+        options = {
+          resource_version: '0'  # Fetch from API server.
+        }
+        if ENV['K8S_NODE_NAME']
+          options[:field_selector] = 'spec.nodeName=' + ENV['K8S_NODE_NAME']
+        end
+        pods = @client.get_pods(options)
```

Comment:
My familiarity with the ruby client is limited but this is enough to restrict the list of pods to a given node?  I have not seen such syntax with any of the other clients or the REST api

Reply:
It is. E.g. https://github.com/kubernetes/client-go/issues/410
It is also recommended by the Google Kubernetes team. @wojtek-t @krzysied is there any reference you can share?

Reply:
There's https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#list-pod-v1-core
```
fieldSelector | A selector to restrict the list of returned objects by their fields. Defaults to everything.
```

Reply:
fieldSelector is widely used in k8s codebase.
As an example, kubelet is doing exactly that (list + watch of its pods) and is using fieldSelector
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L456
[obviously not in ruby]